### PR TITLE
Add delete target endpoint

### PR DESCRIPTION
**Pull request by jackrootstrap**
## Delete target endpoint

* [Trello Card #6-2 Delete targets ](https://trello.com/c/Iy7bgEdl/6-2-as-a-user-i-should-be-able-to-delete-a-previously-created-target-so-its-no-longer-available-in-the-map)
---
#### Description:

* Delete target endpoint

#### Notes:

* Create endpoint for delete a target
* Add request and acceptance specs

### CODE REVIEW

File: app/controllers/api/v1/targets_controller.rb [hunk 1]
```diff
@@ -11,12 +11,20 @@ def create
         @target = targets.create!(target_params)
       end
 
+      def destroy
+        target.destroy!
```
Comment (sebastiancaraballo) [inline app/controllers/api/v1/targets_controller.rb:15]: It'd be nice to return a message like `Target was successfully deleted` or something similar

File: spec/requests/api/v1/targets/destroy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        let(:target) { create(:target) }
+
+        before { subject }
+
+        it 'returns bad request status' do
+          expect(response).to have_http_status(:bad_request)
+        end
+
+        it 'returns an error message' do
+          expect(errors).to include I18n.t('api.errors.target_not_found')
```
Comment (sebastiancaraballo) [inline spec/requests/api/v1/targets/destroy_spec.rb:59]: This should be on the docs I think

Comment (jackrootstrap) [reply spec/requests/api/v1/targets/destroy_spec.rb:59]: I will add this case to the acceptance spec!

**Review by sebastiancaraballo — APPROVED** [review]
💯

File: spec/acceptance/target_spec.rb [hunk 3]
```diff
@@ -41,4 +42,29 @@
...
+
+          expect(status).to eq 200
+        end
+      end
+
+      context 'when target does not belong to user' do
+        let(:new_target) { create(:target) }
+        let(:request) { { id: new_target.id } }
+
+        example 'Delete' do
```
Comment (JulianPasquale) [inline spec/acceptance/target_spec.rb]: Should we rename this `example` to clarify that actually the target is not being deleted?

File: spec/requests/api/v1/targets/destroy_spec.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          expect(response).to have_http_status(:success)
+        end
+
+        it 'returns success message' do
+          subject
+
+          expect(json['message']).to eq I18n.t('api.target.deleted')
+        end
+
+        it 'destroy a target record on the DB' do
```
Comment (JulianPasquale) [inline spec/requests/api/v1/targets/destroy_spec.rb]:
```suggestion
        it 'deletes the target record from the DB' do
```

**Review by JulianPasquale — APPROVED** [review]

**Pull request merged**